src/components/NavButtonWithTooltip/index.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from "react";
+import { Tooltip, TooltipProps } from "antd";
+
+import {
+    NAV_BAR_TOOLTIP_OFFSET,
+    TOOLTIP_COLOR,
+    TOOLTIP_DELAY,
+} from "../../constants";
+import NavButton, { NavButtonProps } from "../NavButton";
+
+interface TooltipText {
+    default: string;
+    disabled?: string;
+}
+interface NavButtonWithTooltipProps extends NavButtonProps {
+    tooltipText?: TooltipText;
+    tooltipPlacement?: TooltipProps["placement"];
+}
+
+const NavButtonWithTooltip: React.FC<NavButtonWithTooltipProps> = ({
+    tooltipText = { default: "" },
+    titleText,
+    tooltipPlacement,
+    buttonType,
+    icon,
+    clickHandler,
+    isDisabled = false,
+    ...props
+}) => {
+    const getTooltipRenderText = () => {
+        return isDisabled && tooltipText.disabled
+            ? tooltipText.disabled
+            : tooltipText.default;
+    };
+
+    const [tooltipRenderText, setTooltipRenderText] =
+        useState(getTooltipRenderText);
+    const [tooltipVisible, setTooltipVisible] = useState(false);
+
+    // The conditional below prevents a flicker when tooltip text changes faster than tooltip can hide
+    useEffect(() => {
+        if (!isDisabled || (!tooltipVisible && isDisabled)) {
+            setTooltipRenderText(getTooltipRenderText());
+        }
```

Comment:
this is personal preference, but as this is written now, I need to read that comment in order to parse what this conditional is doing. It also seems like you're checking `!isDiasbled/isDisabled` both here and in get the render text, and I think it's easier to read to just show in this function what you do when isDisabled is changed 

```suggestion
       if (tooltipVisible) {
           return
       }
        if (isDisabled)) {
            setTooltipRenderText(tooltipText.disabled || "");
        } else {
            setTooltipRenderText(tooltipText.default);
        }
```

Reply:
Note, I wouldn't commit this directly, I wrote it in the GitHub editor so it might break things

Reply:
Only needed to fix one bracket, but with the default you suggested below we can actually get rid of most of this